spec/system/composer/category_templates_spec.rb

```diff
@@ -160,6 +160,77 @@
     sign_in user
   end
 
+  describe "discard draft modal" do
+    it "does not show the modal if there is no draft on a topic without a template" do
+      category_page.visit(category_no_template)
+      category_page.new_topic_button.click
+      composer.close
+      expect(composer).to be_not_opened
+    end
+
+    it "shows the modal if there is a draft on a topic without a template" do
+      category_page.visit(category_no_template)
+      category_page.new_topic_button.click
+      composer.fill_content("abc")
+      composer.close
+      expect(composer).to be_opened
+      expect(composer).to have_discard_draft_modal
+    end
+
+    it "does not show the modal if there is no draft on a topic with a topic template" do
+      category_page.visit(category_topic_template)
+      category_page.new_topic_button.click
+      composer.close
+      expect(composer).to be_not_opened
+    end
+
+    it "shows the modal if there is a draft on a topic with a topic template" do
+      category_page.visit(category_topic_template)
+      category_page.new_topic_button.click
+      composer.append_content(" some more content")
+      composer.close
+      expect(composer).to be_opened
+      expect(composer).to have_discard_draft_modal
+    end
+
+    it "does not show the modal if on a topic with a form template" do
+      category_page.visit(category_with_template_1)
+      category_page.new_topic_button.click
+      composer.close
+      expect(composer).to be_not_opened
+    end
+
+    context "when the default template has a topic template" do
+      SiteSetting.default_composer_category =
+        (
+          if SiteSetting.general_category_id != -1
+            SiteSetting.general_category_id
+          else
+            SiteSetting.uncategorized_category_id
+          end
+        )
```

Comment:
I don't quite understand this... what toggles the `general_category_id` site setting here?

Reply:
When the bug was correctly repro-d, it seemed to only occur when the default category had a topic template present. So I wanted to add a spec for that situation. 

Depending on the testing environment it seems sometimes `uncategorized_category_id` is the default category and other times it could be `general_category_id` so I'm just checking to see which one is set to find the `default_composer_category`. I am then adding `topic_template: "Testing"` to the `default_category`.

I hope that makes sense, let me know if you'd like me to further clarify.

Reply:
Hmm... ok. I would recommend one of two things: 

- add two test cases, explicitly setting up the settings in each (so you can avoid unreliable test behaviour)
- see if you can track down what is causing the different setting values in different test runs/environments? This may be a rabbit hole, though.

Reply:
Feel free to merge this @keegangeorge to fix the issue. The test cases above can be handled in a future PR.